scipy/spatial/setup.py

```diff
@@ -2,13 +2,28 @@
 import glob
 
 
+def pre_build_hook(build_ext, ext):
+    from scipy._build_utils.compiler_helper import (
+        set_cxx_flags_hook, try_add_flag, try_compile, has_flag)
+    cc = build_ext._cxx_compiler
+    args = ext.extra_compile_args
+
+    set_cxx_flags_hook(build_ext, ext)
+
+    if cc.compiler_type == 'msvc':
+        args.append('/EHsc')
```

Comment:
does this deserve a comment? I have no idea what that does. In contrast, there is a comment for the `-fvisibility=hidden` flag below, which is substantially clearer on its own already.

Reply:
As I understand the MSVC documentation it enables standard C++ exception handling and asserts that functions with C linkage (i.e. those declared ```extern "C"```) with not throw an exception. Which is reasonable, they should never do that.

Reply:
+1 for `/EHsc`. If it's helpful to make that the default in `numpy.distutils`, that would be fine I think. (but not worth too much effort unless it's helpful in the short term).

It's also the Meson default: https://mesonbuild.com/Builtin-options.html#compiler-options.

Reply:
I've added a comment for what this does. Also FWIW, `/EHsc` is recommended in the `pybind11` build examples as well.